src/components/notebook/Editor.tsx

```diff
@@ -0,0 +1,155 @@
+import { cn } from "@/lib/utils";
+import { KeyBinding } from "@codemirror/view";
+import * as Dialog from "@radix-ui/react-dialog";
+import { tables } from "@runt/schema";
+import { Maximize2, Minimize2 } from "lucide-react";
+import { useState } from "react";
+import { Button } from "../ui/button";
+import { CodeMirrorEditor } from "./codemirror/CodeMirrorEditor";
+
+// TODO: use radix modal primitives for this to make state management easier
```

Comment:
Did you mean to leave this todo comment here and not in the modal setup (where those primitives are now used)?